server/self_protection.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2021 TiKV Project Authors.
```

Comment:
As the main struct name is `APIRateLimiter`, how about change file name to rate limiter too?